src/utils.ts

```diff
@@ -1,89 +1,124 @@
 import Web3 from 'web3';
 import { AVAILABLE_TLDS } from './constants';
+import { Utils } from './types';
+import { hash as namehash } from 'eth-ens-namehash';
+import { keccak256 } from 'js-sha3';
 
-/**
- * Checks if the contract in the given address has the given method 
- * 
- * @param web3 - Web3 instance 
- * @param contractAddress - address of the contract to check 
- * @param signatureHash - keccak256 of the method signature
- * 
- * @returns
- * true if method exists, false if not
- */
-export const hasMethod = async(web3: Web3, contractAddress: string, signatureHash: string) => {
-  const code = await web3.eth.getCode(contractAddress);
-  return code.indexOf(signatureHash.slice(2, signatureHash.length)) > 0;
-}
-
-/**
- * Checks if the given web3 instance has accounts to sign txs
- * 
- * @param web3 - Web3 instance 
- * 
- * @returns
- * true if has accounts, false if not
- */
-export const hasAccounts = async(web3: Web3) => {
-  let accounts = [];
-  try {
-    accounts = await web3.eth.getAccounts();
-  } catch {
-    return false;
+export default class implements Utils {
+  
+  constructor() { }
+  
+  /**
+   * Checks if the contract in the given address has the given method 
+   * 
+   * @param web3 - Web3 instance 
+   * @param contractAddress - address of the contract to check 
+   * @param signatureHash - keccak256 of the method signature
+   * 
+   * @returns
+   * true if method exists, false if not
+   */
+  async hasMethod(web3: Web3, contractAddress: string, signatureHash: string): Promise<boolean>{
+    const code = await web3.eth.getCode(contractAddress);
+    return code.indexOf(signatureHash.slice(2, signatureHash.length)) > 0;
   }
-  return accounts.length > 0;
-}
 
-/**
- * Validates the given label
- * 
- * @param label - label to validate
- * 
- * @returns
- * true if valid, false if not
- */
-export const validLabel = (label: string) => {
-  return label && !label.match('[^a-z0-9]');
-}
-
-/**
- * Validates the given domain syntax
- * 
- * @param domain - domain to validate
- * 
- * @returns
- * true if valid, false if not
- */
-export const validDomain = (domain:string) => {
-  const labels = domain.split('.');
+  /**
+   * Checks if the given web3 instance has accounts to sign txs
+   * 
+   * @param web3 - Web3 instance 
+   * 
+   * @returns
+   * true if has accounts, false if not
+   */
+  async hasAccounts(web3: Web3): Promise<boolean> {
```

Comment:
Same as `getCode`. We could get the `getAccounts` function to allow different libs to use it.